### PR TITLE
Add .svg to select_jinja_autoescape

### LINKED ISSUE #4831: Automatically escape text in .svg files
## Summary

Currently, autoescaping is enabled for all templates ending in `.html`, `.htm`, `.xml` as well as `.xhtml` when using `render_template()` (https://flask.palletsprojects.com/en/2.2.x/templating/).

As SVG files are very similar in nearly all aspects to XML files, `.svg` should also be autoescaped.

## Problem

In this example, trying to insert a title into an SVG to render, an error will occur if the title contains unescaped text such as an ampersand (eg. "TEST & EXAMPLE")

In a `main.svg` template file:

```svg
<svg xmlns='http://www.w3.org/2000/svg'
    xmlns:xlink='http://www.w3.org/1999/xlink' width="250" height="220" viewBox="0 0 250 220">
    <text fill="#ffffff" font-family="Roboto, Segoe UI, Ubuntu, Arial, sans-serif" font-weight="600" font-size="15px">
        {{ title }}
    </text>
</svg>
```

```py
render_template("main.svg", title="TEST & EXAMPLE")
```

image

## Current workarounds

Changing the filename to .xml can be done as a workaround, as well overriding the autoescape option or other methods for toggling autoescape. I'm proposing this as a feature request as I believe it is logical for SVG to work the same way as XML.

### PR DESCRIPTION
**Pull request by DenverCoder1**
Currently, autoescaping is enabled for all templates ending in `.html`, `.htm`, `.xml` as well as `.xhtml` when using `render_template()` (https://flask.palletsprojects.com/en/2.2.x/templating/).

As SVG files are a type of XML file and are similar in nearly all aspects to XML, `.svg` should also be autoescaped.

This adds `.svg` to the list of file extensions autoescaped by default to make it consistent with `.xml`, `.xhtml`, etc.

- fixes #4831

Checklist:

- [x] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
	- *Test removed as deemed unecessary by @davidism*
- [x] Add or update relevant docs, in the docs folder and in code.
- [x] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [x] Add `.. versionchanged::` entries in any relevant code docs.
- [x] Run `pre-commit` hooks and fix any issues.
- [x] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

File: docs/quickstart.rst [hunk 1]
```diff
@@ -436,6 +436,11 @@ Here is a basic introduction to how the :class:`~markupsafe.Markup` class works:
     >>> Markup('<em>Marked up</em> &raquo; HTML').striptags()
     'Marked up » HTML'
 
+.. versionchanged:: 2.2
```
Comment (pgjones) [inline docs/quickstart.rst:439]: I don't think we need to reference this change here

**Review by pgjones — COMMENTED** [review]
Minor comments, but makes sense to me.

File: CHANGES.rst [hunk 2]
```diff
@@ -9,6 +9,8 @@ Version 2.2.3
 
 Unreleased
 
+-   Autoescaping is now enabled by default for ``.svg`` files. Inside
+    templates this behavior can be changed with the ``autoescape`` tag.
```
[the remark this answers is not in the document]
Comment (davidism) [reply CHANGES.rst:13]: I usually reference the issue, not the PR, but either is fine.

File: tests/test_templating.py [hunk 3]
```diff
@@ -114,6 +114,18 @@ def test_escaping_without_template_filename(app, client, req_ctx):
     assert flask.render_template("mail.txt", foo="<test>") == "<test> Mail"
 
 
+def test_escaping_svg(app, client):
```
Comment (davidism) [inline tests/test_templating.py:117]: Don't need this test, it's just testing that Jinja works. At most, you wouuld test that `app.select_jinja_autoescape` returned `True` for a `.svg` filename.

**Comment by DenverCoder1** [comment]
@davidism @pgjones All reviews are resolved, any updates on this?

**Comment by pgjones** [comment]
Thank you.

**Comment by davidism** [comment]
This got merged into main instead of 2.2.x. Make sure you open a PR against the correct target branch, it's easy to miss. I'll cherry pick the commits to 2.2.x.

**Comment by davidism** [comment]
79032ca5f1c4747e5aeaa193bdeb2e4eae410ea6

**Pull request merged**